### ISSUE TITLE
TFBertPreTrainingLoss has something wrong

### ISSUE THREAD

**Issue by ultimatedaotu** [huggingface/transformers#13046]
## Environment info

- `transformers` version:
- Platform:
- Python version: 3.8
- PyTorch version (GPU?):
- Tensorflow version (GPU?): 2.5
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

## Information

Model I am using (Bert, XLNet ...):  TFBertPreTrainingLoss 

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

1. construct some inputs of MLM task
2. call  TFBertForMaskedLM
3. while computing loss, something wrong happened.

## Expected behavior

```python
masked_lm_loss = loss_fn(y_true=masked_lm_labels, y_pred=masked_lm_reduced_logits)
next_sentence_loss = loss_fn(y_true=next_sentence_label, y_pred=next_sentence_reduced_logits)
masked_lm_loss = tf.reshape(tensor=masked_lm_loss, shape=(-1, shape_list(next_sentence_loss)[0]))
masked_lm_loss = tf.reduce_mean(input_tensor=masked_lm_loss, axis=0)
```

The number of masked_labels is uncertain ，thus ops of "reshape" is unsuitable. Why not calculate the total loss of batches?

**Comment by Rocketknight1** [comment]
Hi, thanks for the issue but I'll need a little more info to investigate! Do you encounter an error when you run the code, or do you believe the outputted loss is incorrect? If you encounter an error, can you paste it here? If the loss is incorrect, can you upload a sample batch of data (e.g. a pickled dict of Numpy arrays) that gets different loss values on the PyTorch versus the TF version of the model? 

All of that will help us track down the problem here. Thanks for helping!
